lib/core/util.c

```diff
@@ -794,3 +794,13 @@ const char *h2o_npn_protocols = NPN_PROTOCOLS_CORE "\x08"
                                                    "http/1.1";
 
 uint64_t h2o_connection_id = 0;
+
+extern void h2o_http2_wbuf_buffer_prototype__dispose(void);
+extern void h2o_mem_pool_allocator__dispose(void);
+
+void h2o_tls_data_dispose(void)
+{
+    h2o_mem_pool_allocator__dispose();
+    h2o_http2_wbuf_buffer_prototype__dispose();
```

Comment:
I appreciate your effort to keep the modularization.

But for this case, I wouldn't mind changing the linkage of static thread-local variables to extern (and releasing them directly in this function), considering the fact that we already have `h2o_socket_buffer_prototype`.

For example, you can rename `mempool_allocator` to `h2o_mem_pool_allocator` and call `h2o_mem_dispose_recycle(&h2o_mem_pool_allocator)` here. The same goes to `wbuf_buffer_prototype` (which could be renamed to `h2o_http2_wbuf_buffer_prototype`.

Reply:
yes, fixed.
but only `extern` here, did not put `extern` in public header file.